pkg/sql/codegen/pai/codegen.go

```diff
@@ -18,34 +18,76 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"regexp"
+	"strings"
 	"text/template"
 
+	"sqlflow.org/gomaxcompute"
+
 	"sqlflow.org/sqlflow/pkg/sql/codegen/tensorflow"
 	"sqlflow.org/sqlflow/pkg/sql/ir"
 )
 
 const entryFile = "entry.py"
 
+func getTableFromSelect(dataSource, trainSelect string) (string, string, error) {
+	// FIXME(typhoonzero): copied from tensorflow/codegen.go, should remove this and use the temp table
+	// in the workflow
+	fromRegex, err := regexp.Compile("FROM[\\s\\n]+([\\w\\.]*)")
```

Comment:
Maybe we need a strict check to make sure the `trainSelect` should be `SELECT * FROM tmp_table TO TRAIN...`, the standard sql statement should not contain any `WHERE/LIMIT/...`.

Reply:
Will fix this in next PR